WordPress/WordPressTest/ReaderPostTest.m

```diff
@@ -4,13 +4,30 @@
 @import WordPressShared;
 
 @interface ReaderPostTest : XCTestCase
+
+@property (nonatomic, strong) id<CoreDataStack> coreDataStack;
+
 @end
 
 @implementation ReaderPostTest
 
+@synthesize coreDataStack = coreDataStack;
+
+- (void)setUp
+{
+    [super setUp];
+    self.coreDataStack = [[TestContextManager alloc] init];
+}
+
+- (void)tearDown
+{
+    [super tearDown];
+    self.coreDataStack = nil;
```

Comment:
Nitpick: [We plan to stop calling `super` in the life-cycle overrides.](https://github.com/wordpress-mobile/WordPress-iOS/issues/17569):

> There's nothing wrong in calling `super` in setup or teardown methods, but it's unnecessary because `XCTestCase` doesn't implement any functionality in there, making calling `super` redundant.
>
> Note that I couldn't find any _explicit_ proof in the docs to back my assertion. But, I'll point you towards [the life-cycle documentation page](https://developer.apple.com/documentation/xctest/xctestcase/understanding_setup_and_teardown_for_test_methods) to see that there is no mention of calling the superclass. 

```suggestion
    self.coreDataStack = [[TestContextManager alloc] init];
}

- (void)tearDown
{
    self.coreDataStack = nil;
```

We could keep them, just for consistency with the current state of things, although I'd argue against it because it only adds extra work to our future selves. 

If we decided to keep them, what do you think of moving the `[super tearDown]` call to the end of the method? That's how I used to see it done elsewhere in the past: first cleanup our code, then let `super` do it's work.

```suggestion
    [super setUp];
    self.coreDataStack = [[TestContextManager alloc] init];
}

- (void)tearDown
{
    self.coreDataStack = nil;
    [super tearDown];
```

Reply:
I don't really have a strong opinion on this. For me, I think it's a habit that I developed over the years - always call super at beginning of the function, unless the API doc explicitly says don't call it or right placement of super call. I can definitely remove those super calls added in this PR due to my personal habit 😄 

If this is something we'd like to do going forward, maybe it's worth considering adding a swiftlint rule for it, like:
```yaml
prohibited_super_call:
  included:
    - setUp()
    - tearDown()
```